test/selftest_suite/selftest_suite.c

```diff
@@ -74,7 +74,7 @@ void test_selftest_suite__bool(void)
 
 void test_selftest_suite__ptr(void)
 {
-	const char *actual = "expected";
-	cl_assert_equal_p(actual, actual); /* pointers to same object */
-	cl_assert_equal_p(&actual, actual);
+	void *p1 = (void *)0x1, *p2 = (void *)0x2;
```

Comment:
I think that technically this is UB. http://www.cs.man.ac.uk/~pjj/cs211/c_rationale/node12.html

Reply:
I guess you refer to the following quote, right?

>  Since pointers and integers are now considered incommensurate, the only integer that can be safely converted to a pointer is the constant 0. The result of converting any other integer to a pointer is machine dependent. 

And the following makes clear that even the comparison of invalid pointers is UB:

> Regardless how an invalid pointer is created, any use of it yields undefined behavior. Even assignment, comparison with a null pointer constant, or comparison with itself, might on some systems result in an exception. 

One alternative would be to use `intptr_t` instead, but that's an optional extension to C99, only. So I'm inclined to just leave this as-is, as it works just fine in practice on most systems, unless you have a better idea.

Reply:
Yep, I think that's the pragmatic approach. It is after all, the tests for the tests, so a little questionably UB behavior probably won't hurt too bad.